Woo Installer: Landing page redesign

allilevine opened the pull request:
#### Changes proposed in this Pull Request

* Add new landing page component to /woocommerce-installation flow.
* Add new `CtaSection` component.
* Change fixed navigation component to take a `contentRef` prop and show/hide the right content depending on whether it's also visible on the page.

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Checkout this branch and click on the "WooCommerce" nav menu item on an Atomic site that doesn't have Woo installed.
* Verify that the new landing page displays.
* Check that clicking "Set up my store!" still works to install Woo.
* Check that when you scroll past the CTA button, it appears in the header.
* Check that other pages using the fixed navigation component (like /plugins) display correctly.
* Check that you get a notice if you try to start this flow without confirming your email address.

Before
<img src="https://user-images.githubusercontent.com/811776/141930667-a17afe15-1848-43b8-ba8f-5f2750f69922.png">

After
<img src="https://user-images.githubusercontent.com/811776/141930655-a08ad8c3-d858-4204-b5be-040b55177021.png">



<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #57419

allilevine committed:
Add WoopLandingPage component

allilevine committed:
Add WoopLandingPage component

allilevine committed:
Add cta header component

allilevine committed:
Add cta header component

allilevine committed:
Remove extra comment

allilevine committed:
Remove extra comment

allilevine committed:
Style landing page

allilevine committed:
Style landing page

lsl commented:
Looks great! You can pull the images from https://woocommerce.com's homepage, e.g. https://woocommerce.com/wp-content/themes/woo/images/homepage-hero/image01.jpg.

Not sure where this "woo" theme is located. Could probably ask around if it helps but pulling them down from the site seems easy enough.

allilevine committed:
Hide/Show Navigation Button when scrolling past the CTA Button

allilevine committed:
Re-organize components and use Types

allilevine committed:
Add SetupNotices to the new landing page

allilevine committed:
Add some responsive CSS and remove mixin

allilevine committed:
Remove style import and use rem

Rolilink committed:
Adds WaveMasonry component to the woop landing page

Rolilink committed:
Rendering only images

Rolilink committed:
changes cta-header to 1 column layout on mobile

lsl commented on client/my-sites/woocommerce/dashboard/required-plugins-install-view.jsx:
```diff
@@ -597,7 +598,8 @@ class RequiredPluginsInstallView extends Component {
 		if ( ! hasPendingAT && 'CONFIRMING' === engineState ) {
-			return this.renderConfirmScreen();
+			return <WoopLandingPage startSetup={ this.startSetup } />;
+			//return this.renderConfirmScreen();
```
```suggestion
```

lsl replied:
Should also delete the function.

allilevine replied:
Thanks for catching that, it reappeared from a merge conflict 😂

lsl reviewed (approved):
LGTM, added screenshots to PR, quick checked mobile, is there anything preventing this from merging?

lsl commented on client/components/masonry-wave/index.jsx:
```diff
@@ -0,0 +1,91 @@
+	const { image } = props;
+
+	return (
+			<img src={ image?.src } alt={ image?.alt ?? 'alt text' } className="masonry-wave__element" />
```
```suggestion
		<img src={ image?.src } alt={ image?.alt ?? 'alt text' } className="masonry-wave__element" />
```

lsl committed:
Update client/components/masonry-wave/index.jsx

lsl commented:
Committed a lint fix that was blocking the CI build, things should pass now but lets clean up the old render function too.

pablinos commented on client/components/masonry-wave/index.jsx:
```diff
@@ -0,0 +1,91 @@
+	};
+}
+
+function useWindowDimensions() {
```
I think it might be worth using the [automattic/viewport](https://github.com/Automattic/wp-calypso/tree/trunk/packages/viewport) package in here

Rolilink replied:
There is a react helper for this one will probably use [this one](https://github.com/Automattic/wp-calypso/tree/trunk/packages/viewport-react)

pablinos commented on client/components/cta-header/style.scss:
```diff
@@ -0,0 +1,5 @@
+@import '@automattic/onboarding/styles/mixins';
+
+h4 {
```
I'm guessing this exists as we can't include the mixins in the styled components? If that's the case, should we put the rest of the styles in here? But we are already mixing styled components and SCSS elsewhere in signup.

Do we know if the move is towards or away from SCSS in general?

allilevine replied:
It seems like the newer components are using styled components / `emotion`. I realized I could use `font-family: Recoleta;` and it works, so I've deleted this file.

Rolilink replied:
> Do we know if the move is towards or away from SCSS in general?

I think I read something from the calypso team about this, but I think that is a good idea to ask around!

pablinos commented on client/components/fixed-navigation-header/index.tsx:
```diff
@@ -55,16 +56,53 @@ interface Props {
-				<ActionsContainer>{ children }</ActionsContainer>
+				<ActionsContainer
+					ref={ actionsRef }
+					style={ { visibility: `${ contentRef ? 'hidden' : 'visible' }` } }
```
Could having this property managed by a prop and then manipulated via the DOM API in `handleScroll` cause some problems? It seems ok, but I wondered if it might be safer to trigger `handleScroll` in the `useEffect` to set the initial value.

Also, I'm not sure if it's working as intended. When I scroll nothing seems to get hidden, or become visible. I think the intention is that the button at the top is hidden until the button in the main content is scrolled out of view.

https://user-images.githubusercontent.com/96462/141977229-0ee64b6b-003c-451a-bc4c-d3a7fe583d75.mov

allilevine replied:
> Could having this property managed by a prop and then manipulated via the DOM API in `handleScroll` cause some problems? It seems ok, but I wondered if it might be safer to trigger `handleScroll` in the `useEffect` to set the initial value.

That makes more sense 👍 I made the change.

> Also, I'm not sure if it's working as intended. When I scroll nothing seems to get hidden, or become visible. I think the intention is that the button at the top is hidden until the button in the main content is scrolled out of view.
> Screen.Recording.2021-11-16.at.11.25.45.mov

I think it's working correctly now:

https://user-images.githubusercontent.com/1689238/142062714-632eca04-aa7e-4279-8d21-8b4ff3dff950.mov

pablinos commented on client/components/masonry-wave/index.jsx:
```diff
@@ -0,0 +1,91 @@
+			setWindowDimensions( getWindowDimensions() );
+		}
+
+		window.addEventListener( 'resize', handleResize );
```
The viewport package has helpers to subscribe to changes in size, but we might want to add some throttling/debouncing to this.

Rolilink replied:
Good catch in here but will probably use https://github.com/Automattic/wp-calypso/tree/trunk/packages/viewport-react !

pablinos reviewed (commented):
Looking good. I realise this is still in draft, but I decided to stick my nose in and share some thoughts anyway!

I think we might want to adjust the margin/padding around the edges and in between the text and images. 

We may also want to trigger the mobile view a little earlier, as the elements get a little cramped.

<img width="400" alt="image" src="https://user-images.githubusercontent.com/96462/141935559-b33e2625-4125-4e32-89f8-f9d5da545f54.png">

<img width="400" alt="image" src="https://user-images.githubusercontent.com/96462/141935811-f3a0b847-4d04-42c6-99d0-80c842e3361a.png">

<img width="400" alt="image" src="https://user-images.githubusercontent.com/96462/141935863-1431b5e3-ec93-4acc-8135-89234200f74b.png">

I've also noticed that the page title doesn't update on this page, so it's set to the title of the previous page. That's not introduced by this change, but might be worth fixing.

pablinos commented:
> LGTM, added screenshots to PR, quick checked mobile, is there anything preventing this from merging?

Shows what happens when you get sidetracked halfway through a review. I think there are some bits we might want to tidy up in here, but it's definitely close.

retrofox commented on client/my-sites/woocommerce/woop/woop-landing-page.jsx:
```diff
@@ -0,0 +1,36 @@
+import { Button } from '@automattic/components';
+import { translate } from 'i18n-calypso';
+import Image01 from 'calypso/assets/images/woocommerce/image01.jpeg';
```
should we rename the files with something more descriptive? or maybe create a folder?

retrofox replied:
please feel free to ignore this suggestion :-)

Rolilink replied:
✅ We already have a folder for this one under `assets/images/woocommerce` but went ahead and changed the name to `woop-cta-imageXX` that is more descriptive.

retrofox commented on client/my-sites/woocommerce/woop/woop-landing-page.jsx:
```diff
@@ -0,0 +1,36 @@
+
+const images = [ { src: Image01 }, { src: Image05 }, { src: Image02 }, { src: Image04 } ];
+
+const WoopLandingPage = ( props ) => {
+	const { startSetup } = props;
```
```suggestion
const WoopLandingPage = ( { startSetup } ) => {
```

retrofox replied:
feel free to ignore it too

allilevine replied:
That's a good one, I made the change 👍

allilevine committed:
Clean up merge conflicts

allilevine committed:
Clean up landing-page prop

allilevine commented on client/components/masonry-wave/style.scss:
```diff
@@ -0,0 +1,23 @@
+	grid-auto-rows: minmax(min-content, max-content);
+	column-gap: 0.75rem;
+	.masonry-wave__column {
+		&:nth-child(even) {
```
@Rolilink Looks like we're getting a console warning, in case it's related to this:
`The pseudo class ":first-child" is potentially unsafe when doing server-side rendering. Try changing it to ":first-of-type".`

Rolilink replied:
@allilevine This warning in here is actually coming from the Breadcrumb component in[ here](https://github.com/Automattic/wp-calypso/blob/trunk/client/components/breadcrumb/index.tsx#L19). I will log an issue to fix this one since I will need some time to test that the component is still working fine after adding the change.

allilevine commented on client/components/masonry-wave/index.jsx:
```diff
@@ -0,0 +1,91 @@
+	return (
+		<div className="masonry-wave__column">
+			{ images.map( ( image ) => (
+				<MasonryWaveElement image={ image } />
```
@Rolilink I'm seeing some console warnings about missing `key` props on `MasonryWave` and `MasonryWaveColumn`:
`Each child in a list should have a unique "key" prop.`

Rolilink replied:
✅  Added the keys for this component

allilevine committed:
Clean up spacing, scroll

allilevine committed:
Add WooCommerce title

Rolilink committed:
cta-header uses css-grid 1 column layout and centers content on mobile

Rolilink committed:
adds @automattic/viewport-react and fixes key related warnings

- Adds @automattic/viewport-react to the MasonryWave component to conditionally render different layouts on mobile and web.
- Fixes missing key warnings on the MasonryWave component.

Rolilink committed:
renaming cta images to a more descriptive name | removing unused cta images

retrofox reviewed (approved):
Thanks for addressing the feedback. LGTM. :-)
